src/Serialize.h

```diff
@@ -33,6 +33,101 @@ namespace drafter {
 
     refract::Registry& GetNamedTypesRegistry();
 
+    template<typename T>
+    struct SectionInfo {
+        typedef SectionInfo<T> Type;
+        typedef T SectionType;
+        typedef snowcrash::SourceMap<SectionType> SourceMapType;
+
+        const SectionType& section;
```

Comment:
I start with ParseResult, but after few tests I create new structure. 
Mainly ParseResult contains values instead of reference (it mean - copy value while creating ParseResult instance), I use just reference to avoid copy.  Additionaly ParseResult contain Annotation.